rainier-core/src/main/scala/com/stripe/rainier/sampler/LongestBatch.scala

```diff
@@ -0,0 +1,97 @@
+package com.stripe.rainier.sampler
+
+/**
+  * Compute a distribution of number of leapfrog steps required to make
+  * a u-turn using a tuned value of the step size eps
+  * @param l0 the initial number of leapfrog steps
+  * @param eps the tuned leapfrog step size
+  */
+object LongestBatch {
+
+  /**
+    * Calculate the longest-step size until a u-turn
+    */
+  def longestBatch(params: Array[Double],
```

Comment:
would it be accurate to name this `stepUntilUTurn`?

Reply:
Yep, I just used the function name from the paper.